Mathlib/AlgebraicTopology/SimplicialSet/Degenerate.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.Basic
+
+/-!
+# Degenerate simplices
+
+Given a simplicial set `X` and `n : ℕ`, we define the sets `X.degenerate n`
+and `X.nonDegenerate n` of degenerate or non-degenerate simplices of dimension `n`.
+
+## TODO (@joelriou)
+
+* `SSet.exists_nonDegenerate` shows that any `n`-simplex can be written
+as `X.map f.op y` for some epimorphism `f : [n] ⟶ [m]` and some
+non-degenerate simplex `y`. Show that `f` and `y` are unique.
+
+-/
+
+universe u
+
+open CategoryTheory Simplicial Limits Opposite
+
+namespace SSet
+
+variable (X : SSet.{u})
+
+/-- An `n`-simplex of a simplicial set `X` is degenerate if it is in the range
+of `X.map f.op` for some morphism `f : [n] ⟶ [m]` with `m < n`. -/
+def degenerate (n : ℕ) : Set (X _[n]) :=
+  setOf (fun x ↦ ∃ (m : ℕ) (_ : m < n) (f : ([n] : SimplexCategory) ⟶ [m]),
+    x ∈ Set.range (X.map f.op))
```

Comment:
Not blocking, but I'm curious what the convention is for taking inputs in the `SimplexCategory`. i.e., when should we use `(m : ℕ) (f : [n] ⟶ [m])` vs. `(m : SimplexCategory) (f : [n] ⟶ m)`?

This has some small implications for pretty-printing these notations, but otherwise it seems to me that any convention is just as good here.

Reply:
In this particular case, as there is an inequality, I thought it was more convenient to write it as `m < n` rather than `m.len < n`. It does not matter here as this is part of the internals of the API and I provide some lemmas like `degenerate_eq_iUnion_range_σ` which gives more explicit characterizations. Otherwise, for general definitions and lemmas, if it makes sense for `m : SimplexCategory`, it is advisable to use `SimplexCategory` rather than `ℕ`.

Reply:
For example, below, for `isIso_of_nonDegenerate`, I have used `m : SimplexCategory` and shown that `f : [n]  ⟶ m` was an isomorphisms, which allows to show that `m` is equal to `[n]`, and then `f` is actually the identity of `[n]`.

Reply:
That's helpful, thanks. I suppose `m : SimplexCategory` is strictly more general since the term need not be of the form `SimplexCategory.mk _`.

Incidentally, `m.len` cannot be typed in a subscript. You can write `[n]₍ₗₑₙ ₘ₎`, but this only happens to work because all of the letters involved can be subscripted. One can always fall back to writing out the term where the notation falls down, but do you think it would be worth revisiting the choice of subscripts for this notation while the brackets are in question?

Reply:
I would think that in most applications, the subscript will be a litteral 1-digit natural number, no?

Reply:
Yes, my assumption (and experience so far) is that `[m]ₙ₊₁` is about as complex an expression as we would need for the truncation level. If that seems reasonable, I think I will avoid further cluttering the notation discussion with subscript questions.